tests/integration/test_flow_assignment.py

```diff
@@ -110,53 +113,65 @@ async def test_flow_assignment(
-        assert active_b.flow_nums == {1, 2}
+    async with start(schd) as log:
+
+        # capture task submissions (prevents real submissions)
+        submitted_tasks = capture_submission(schd)
+
+        command = "set"
+        do_command = functools.partial(
+            schd.pool.set_prereqs_and_outputs, outputs=['x'], prereqs=[]
+        )
+  
+        # get active foo and bar (which is which doesn't matter)
+        # ("active" here means in the active task pool).
+        active_x, active_y = schd.pool.get_tasks()
+        schd.pool.merge_flows(active_y, schd.pool._get_flow_nums([FLOW_NEW]))
+        assert active_x.flow_nums == {1}
+        assert active_y.flow_nums == {1, 2}
 
         # -----(1. Test active tasks)-----
 
+        # Note this also tests that setting prerequisites 
+
         # By default active tasks keep existing flow assignment.
-        do_command([active_a.identity], flow=[])
-        assert active_a.flow_nums == {1}
+        do_command([active_x.identity], flow=[])
+        assert active_x.flow_nums == {1}
 
         # Else merge existing flow with requested flows.
-        do_command([active_a.identity], flow=[FLOW_ALL])
-        assert active_a.flow_nums == {1, 2}
+        do_command([active_x.identity], flow=[FLOW_ALL])
+        assert active_x.flow_nums == {1, 2}
 
         # (no-flow is ignored for active tasks)
-        do_command([active_a.identity], flow=[FLOW_NONE])
-        assert active_a.flow_nums == {1, 2}
+        do_command([active_x.identity], flow=[FLOW_NONE])
+        assert active_x.flow_nums == {1, 2}
         assert log_filter(
             contains=(
-                f'[{active_a}] ignoring \'flow=none\' {command}: '
-                f'task already has {repr_flow_nums(active_a.flow_nums)}'
+                f'[{active_x}] ignoring \'flow=none\' {command}: '
+                f'task already has {repr_flow_nums(active_x.flow_nums)}'
             ),
             level=logging.ERROR
         )
 
-        do_command([active_a.identity], flow=[FLOW_NEW])
-        assert active_a.flow_nums == {1, 2, 3}
+        do_command([active_x.identity], flow=[FLOW_NEW])
+        assert active_x.flow_nums == {1, 2, 3}
 
         # -----(2. Test inactive tasks)-----
-        if command == 'set':
-            do_command = functools.partial(
-                schd.pool.set_prereqs_and_outputs, outputs=[], prereqs=['all']
-            )
+        do_command = functools.partial(
+            schd.pool.set_prereqs_and_outputs, outputs=[], prereqs=['all']
+        )
 
         # By default inactive tasks get all active flows.
         do_command(['1/a'], flow=[])
         assert schd.pool._get_task_by_id('1/a').flow_nums == {1, 2, 3}
 
+        # set --pre=all should not cause a task to submit in a paused workflow
+        assert len(submitted_tasks) == 0
+        # but triggering it should
+        schd.pool.force_trigger_tasks(['1/a'], [])
+        schd.release_tasks_to_run()
+        assert len(submitted_tasks) == 1
```

Comment:
Please could you put this new bit in a separate test and revert the splitting of `trigger` and `set` from this test; a key feature of this test was that is checks equity between `trigger` and `set` for the same flow nums

Reply:
The reason I split that into separate tests is that this PR changes that equivalence. 

Trigger and set (in a paused workflow) are more similar on master, because a triggered task does not become "active" until the workflow is resumed, so it can be triggered multiple times to absorb multiple flow numbers. On this branch, a triggered task becomes active immediately and so subsequent triggers get ignored.

In light of that, do you still want me to revert that? I suppose we could have a combined test for what is common, and separate tests for the behavior that differs.

Reply:
> In light of that, do you still want me to revert that? I suppose we could have a combined test for what is common, and separate tests for the behavior that differs.

👆 That

Reply:
Done